src/Chapter3.hs

```diff
@@ -971,6 +1101,20 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfTheWeek = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday deriving (Eq, Bounded, Enum)
+
+isWeekend :: DayOfTheWeek -> Bool
+isWeekend d 
+    | d == Saturday = True 
+    | d == Sunday = True
+    | otherwise = False
+
+nextDay :: DayOfTheWeek -> DayOfTheWeek
+nextDay d 
+    | d == maxBound = minBound 
```

Comment:
Nice implementation! And it will automatically work even if you decide to change the `DayOfTheWeek` type to the American system :+1: